pkg/api/og.go

```diff
@@ -0,0 +1,462 @@
+			return textBox, minFontSize
+		}
+	}
+
+	// Fallback - just ellipsis
+	return canvas.NewTextBox(face, "...", availableWidth, 40, canvas.Left, canvas.Center, &canvas.TextOptions{}), minFontSize
+}
+
+func (a *StreamplaceAPI) HandleOGImage(ctx context.Context) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		username := ps.ByName("username")
+		if username == "" {
+			http.Error(w, "Username required", http.StatusBadRequest)
+			return
+		}
+
+		// Check cache first
+		cacheKey := fmt.Sprintf("og_image_%s", username)
+		if cached, found := a.OGImageCache.Get(cacheKey); found {
+			imgData := cached.([]byte)
+			log.Debug(ctx, "OG image cache hit", "username", username, "size_bytes", len(imgData))
+
+			// Set headers
+			w.Header().Set("Content-Type", "image/jpeg")
+			w.Header().Set("Cache-Control", "public, max-age=3600") // Cache for 1 hour
+			w.Header().Set("Content-Length", fmt.Sprintf("%d", len(imgData)))
+			w.Header().Set("X-Cache", "HIT")
+
+			// Write the cached image data
+			code, werr := w.Write(imgData)
+			if werr != nil {
+				log.Error(ctx, "failed to write cached OG image", "username", username, "error", werr)
+				http.Error(w, "Failed to write cached image: "+werr.Error(), http.StatusInternalServerError)
+				return
+			}
+			log.Debug(ctx, "OG image served from cache", "username", username, "code", code)
+			return
+		}
+
+		// Generate the OG image
+		imgData, err := a.generateOGImage(ctx, username)
+		if err != nil {
+			if errors.Is(err, ErrUserNotFound) {
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
+			}
+			log.Error(ctx, "failed to generate OG image", "username", username, "error", err)
+			http.Error(w, "Failed to generate image: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Store in cache
+		a.OGImageCache.Set(cacheKey, imgData, cache.DefaultExpiration)
+		log.Debug(ctx, "OG image generated and cached", "username", username, "size_bytes", len(imgData))
+
+		// Set headers
+		w.Header().Set("Content-Type", "image/jpeg")
+		w.Header().Set("Cache-Control", "public, max-age=3600") // Cache for 1 hour
+		w.Header().Set("Content-Length", fmt.Sprintf("%d", len(imgData)))
+		w.Header().Set("X-Cache", "MISS")
+
+		// Write the image data
+		code, werr := w.Write(imgData)
+
+		if werr != nil {
+			log.Error(ctx, "failed to write generated OG image", "username", username, "error", werr)
+			// if includes "no repo found" in error, return 404
+			if strings.Contains(werr.Error(), "no repo found") {
+				http.Error(w, "User not found", http.StatusNotFound)
+				return
+			}
+			http.Error(w, "Failed to write image: "+werr.Error(), http.StatusInternalServerError)
+			return
+		}
+		log.Debug(ctx, "OG image generated and served", "username", username, "code", code)
+	}
+}
+
+func downloadImage(url string) ([]byte, error) {
+	if url == "" {
+		return nil, errors.New("empty URL provided")
+	}
+
+	resp, err := http.Get(url)
```

Comment:
Should use `aqhttp` here to get our user-agent and stuff, an example: https://github.com/streamplace/streamplace/blob/395eb2f39e8a1befe6036836497089b1a60f7d49/pkg/integrations/discord/send-chat.go#L51-L61